File: Data Structures & Algorithms/sliding-window-maximum/submission-1.java
class Solution {
    public int[] maxSlidingWindow(int[] nums, int k) {
        PriorityQueue<Integer> topK = new PriorityQueue<>(k);
        List<Integer> result = new ArrayList<>();
        for(int i = 0; i < k; ++i){
            topK.offer(-nums[i]);
        }
        result.add(-topK.peek());
        int left = 0;
        for(int i = k; i < nums.length; ++i){
            topK.remove(-nums[left]);
            topK.offer(-nums[i]);
            result.add(-topK.peek());
            left += 1;
        }
        return result.stream().mapToInt(Integer::intValue).toArray();


    }
}

File: Data Structures & Algorithms/products-of-array-discluding-self/submission-1.py
class Solution:
    def productExceptSelf(self, nums: List[int]) -> List[int]:
        #[1, 2, 3, 4, 5]
        result = [1] * len(nums)

        prefix = 1
        for i in range(1, len(nums)):
            result[i] = prefix * nums[i-1]
            prefix = result[i]
        print(result)
        suffix = 1
        for i in range(len(nums) - 2, -1, -1):
            result[i] = result[i] * suffix * nums[i + 1]
            suffix = suffix * nums[i + 1]
        print(result)

        return result